cores/arduino/stm32/dma.c

```diff
@@ -0,0 +1,315 @@
+      ]0MT   ]M]M  jM@   ]M]     M]P  j       J     j j     4     1        
+      MM]'   M]0P  j]1  .M]M    j]M'  J      j'     ",?     j     1        
+     _]M]    M]0`  jM1 .MNMM,  .]M'   1     .'       11     1    j'        
+     jM]1   jM]@   j]L_]'?M]M__MP'    \     J        1G    J    .'         
+     j]0(   jM]1   "M]P'  "N]M/-      "L__J L________'?L__- *__,'          
+     "-'    "--                                                            
+                                                                           
+----------------------------------------------------------------------------
+
+  ******************************************************************************
+  * @file    dma.c
+  * @author  MCD Application Team, Wi6labs
+  * @brief   dma service.
+  ******************************************************************************
+
+Copyright (c) <2017>, <Wi6labs>
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of the wi6labs nor the
+      names of its contributors may be used to endorse or promote products
+      derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL WI6LABS BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+describe file content here
+-----------------------------------------------------------------------------*/
+#include "stm32_def.h"
+#include "dma.h"
+#include "analog.h"
+
+#ifdef __cplusplus
+ extern "C" {
+#endif
+
+extern ADC_HandleTypeDef AdcHandleTab[ADC_NUM];
+
+static DMA_HandleTypeDef DmaHandleTab[ADC_NUM] = {NULL};
+static bool dmaTab[DMA_NUMBER][CHANNEL_NUMBER];
+
+/**
+  * @brief  This function will init the dma and channel corresponding 
+  *         to the ADC peripheral
+  * @param  pin : the ADC peripheral to use
+  * @retval none
+  */
+void dma_init_adc(ADC_HandleTypeDef *hadc)
+{
```

Comment:
As mentioned DMA PAI should not mention ADC at all. 
instead of dma_init_adc I would rather consider a generic function like dma_allocate to look for the proper DMA instance and channel with a generic (void*) pointer to device. then the function would look for an instance match in the table with pinmap_find_dmaChannel.

Reply:
dma_allocate create. A enum is create, we need it for identify the peripheral.

Reply:
sorry I don't understand your answer. 

My proposal is 
replace : void dma_init_adc(ADC_HandleTypeDef *hadc)
by : void dma_init(void *any_handle_typedef)

so that it can later be used also by other devices than ADC

Reply:
This is what we have done. But in dma_init() we also need an other parameter for know on what peripheral we work (ADC, UART). So the prototype is : 
void dma_allocate(void *periph_handle, Peripheral periph);